### PR TITLE
docs: update build instructions on how to use the docker-yocto-builder

### PR DESCRIPTION
**Pull request by jklare**
Signed-off-by: Jan Klare <jan.klare@bisdn.de>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,13 +7,43 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
 * Disk space and RAM
 
 We recommend a minimum of 8 GiB RAM and 150 GiB free disk space for building a single full
 image.
 
+* A functional Yocto build environment
+
+This can be achieved by either installing and configuring everything as
+documented in the section
+[Building your own build environment](#building-your-own-build-environment), or
```
Comment (hilmarm) [inline README.md:21]:
```suggestion
[Setting up your own build environment](#setting-up-your-own-build-environment), or
```

File: README.md [hunk 2]
```diff
@@ -7,13 +7,43 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
+[Bootstrap build system](#bootstrap-build-system) you have to directly
+exec into the container, while ideally also mounting a local path on your
+file system (e.g. a new folder in your local /tmp) to the default output
+directly inside of the container. This can be achieved by running:
+
+```bash
+docker run -ti -v /tmp/output:/home/builder/output ghcr.io/bisdn/docker-yocto-builder-ci:main bash
+```
+
+## Building your own build environment
```
Comment (hilmarm) [inline README.md:45]:
```suggestion
## Setting up your own build environment
```

File: README.md [hunk 3]
```diff
@@ -7,13 +7,43 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
+[Building your own build environment](#building-your-own-build-environment), or
+by starting from the
+[prebuild Yocto environment](https://github.com/bisdn/docker-yocto-builder) in a
+docker container, as [described below](#using-the-docker-yocto-builder).
+
+## Using the [docker-yocto-builder](https://github.com/bisdn/docker-yocto-builder)
+
+To ease the process of "getting started building BISDN Linux with Yocto", we
+have compiled all dependencies needed by Yocto itself, as well as all additional
+ones for specifically BISDN Linux, into a docker container. This container can
```
Comment (hilmarm) [inline README.md:30]:
```suggestion
have compiled all dependencies needed by Yocto itself, as well as all additional
specific BISDN Linux dependencies, into a docker container. This container can
```

File: README.md [hunk 4]
```diff
@@ -7,13 +7,43 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
+be pulled directly from the GitHub container registry (ghcr)
```
Comment (hilmarm) [inline README.md:31]:
```suggestion
be pulled directly from the GitHub container registry ([ghcr](https://github.blog/2020-09-01-introducing-github-container-registry/))
```

**Review by hilmarm — COMMENTED** [review]
A few suggestions

File: README.md (hunk 4 above)
Comment (jklare) [reply README.md:31]: I don't think documentation (or an intro blog post about it) about ghcr belongs here.

File: README.md (hunk 3 above)
Comment (jklare) [reply README.md:30]: i don't think using 'dependencies' two times in one sentence it good, but i will think about a better way to write this

File: README.md (hunk 2 above)
Comment (jklare) [reply README.md]: reworded to "Installing"

File: README.md (hunk 1 above)
Comment (jklare) [reply README.md]: reworded to "Installing"

File: README.md (hunk 3 above)
Comment (jklare) [reply README.md]: reworded

File: README.md (hunk 4 above)
Comment (hilmarm) [reply README.md]: Maybe some link to what it is. I could not actually find anything satisfying so maybe it's better to just leave it like this.

Comment (jklare) [reply README.md]: Not sure what you mean here?

Comment (jklare) [reply README.md]: updated

Comment (hilmarm) [reply README.md]: Good solution

**Comment by hilmarm** [comment]
Looks good to me

**Review by hilmarm — APPROVED** [review]
lgtm

File: README.md [hunk 5]
```diff
@@ -7,13 +7,57 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
+mkdir /tmp/builder-tmp && chmod 0777 /tmp/builder-tmp
+# start the docker container and mount the folder previously created as /tmp
+docker run -ti -v /tmp/builder-tmp:/tmp ghcr.io/bisdn/docker-yocto-builder-ci:main bash
+```
+
+If you are planning to overwrite the TMPDIR in the conf/local.conf (`TMPDIR =
+"/tmp/${MACHINE}"` by default) as mentioned in the section
+[Configure target machine and cache directory](#configure-target-machine-and-cache-directory),
+you should also consider to mount your local file system path to this new path
+inside of your container.
```
Comment (KanjiMonster) [inline README.md:57]: I would add a link at the end to where you are to continue afterwards, so the reader does not have to backtrack to find out where to continue, e.g. something like this:

```suggestion
inside of your container.

After you have successfully exec'd into the container, you are ready to [bootstrap the build system](#bootstrap-build-system).
```

File: README.md [hunk 6]
```diff
@@ -7,13 +7,57 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
+either have the same UID (1000) as the builder user inside of the container, or
+create the folder beforehand with more permissive access rights.
+This can be achieved by running:
+
+```bash
+# create a folder to be mounted as /tmp inside of the container with very
+# permissive access rights
+mkdir /tmp/builder-tmp && chmod 0777 /tmp/builder-tmp
+# start the docker container and mount the folder previously created as /tmp
+docker run -ti -v /tmp/builder-tmp:/tmp ghcr.io/bisdn/docker-yocto-builder-ci:main bash
```
Comment (KanjiMonster) [inline README.md:50]: Maybe add a simple "does it work" test here touch or so?
```suggestion
docker run -ti -v /tmp/builder-tmp:/tmp ghcr.io/bisdn/docker-yocto-builder-ci:main bash
# (within the container) check that /tmp is writable
[ -w /tmp ] && echo "I can write!"
```

Comment (jklare) [reply README.md:50]: i think this makes it more complicated to understand and imho the description about a writable tmp dir above should be enough for that purpose

File: README.md [hunk 7]
```diff
@@ -7,13 +7,60 @@ The build process takes around 3-4 hours with 8 CPU cores and 8 GiB RAM. A singl
...
 We recommend a minimum of 8 GiB RAM and 150 GiB free disk space for building a single full
 image.
 
+* A functional Yocto build environment
+
+This can be achieved by either installing and configuring everything as
+documented in the section
+[Installing your own build environment](#installing-your-own-build-environment), or
+by starting from the
+[prebuild Yocto environment](https://github.com/bisdn/docker-yocto-builder) in a
```
Comment (hilmarm) [inline README.md:23]:
```suggestion
[prebuilt Yocto environment](https://github.com/bisdn/docker-yocto-builder) in a
```

**Review by hilmarm — CHANGES_REQUESTED** [review]
Past tense

**Review by hilmarm — APPROVED** [review]

**Pull request merged**